### PR TITLE
Consolidate execution context methods.

### PR DESCRIPTION
**Pull request by jeschkies**

### CODE REVIEW

File: commons/src/test/scala/com/mesosphere/usi/async/RetryTest.scala [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+import akka.actor.{Cancellable, Scheduler}
+import com.mesosphere.usi.async.Retry.RetryOnFn
+import com.mesosphere.utils.AkkaUnitTest
+import org.scalatestplus.scalacheck.ScalaCheckPropertyChecks
+
+import scala.collection.mutable
+import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future}
+
+class RetryTest extends AkkaUnitTest with ScalaCheckPropertyChecks {
```
Comment (jeschkies) [inline commons/src/test/scala/com/mesosphere/usi/async/RetryTest.scala]: I also forgot to import these.

**Comment by timcharper** [comment]
Exit status 1 when running "sbt +test:compile"

<details>
<summary>Details</summary>
<pre>
[warn]   class Fixture(debounceReviveInterval: FiniteDuration = 1.seconds, defaultRole: String = &quot;web&quot;) {
[warn]                                                                     ^
[warn] /home/travis/build/mesosphere/usi/core/src/test/scala/com/mesosphere/usi/core/revive/SuppressReviveHandlerTest.scala:30:18: parameter value prefix in method newPodId is never used
[warn]     def newPodId(prefix: String): PodId = PodId(s&quot;prefix-${podIdIndex.incrementAndGet()}&quot;)
[warn]                  ^
[warn] one warning found
[info] Done compiling.
[info] Compiling 2 Scala sources to /home/travis/build/mesosphere/usi/examples/core-hello-world/target/scala-2.12/test-classes ...
[warn] /home/travis/build/mesosphere/usi/examples/core-hello-world/src/test/scala/com/mesosphere/usi/examples/CoreHelloWorldFrameworkTest.scala:110:17: parameter value existingFrameworkId in class Fixture is never used
[warn]   class Fixture(existingFrameworkId: Option[FrameworkID.Builder] = None) {
[warn]                 ^
[warn] one warning found
[info] Done compiling.
[warn] there were two deprecation warnings (since 2.6.0); re-run with -deprecation for details
[warn] three warnings found
[info] Done compiling.
[error] (commons / Test / compileIncremental) Compilation failed
[error] Total time: 79 s (01:19), completed Jun 25, 2020 3:23:46 PM

restoring stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
</pre>
</details>

**Comment by timcharper** [comment]
Exit status 1 when running "Unused imports / scalafix"

<details>
<summary>Details</summary>
<pre>
[warn] one warning found
[info] Done compiling.
[info] Running scalafix on 3 Scala sources
[info] Compiling 2 Scala sources to /home/travis/build/mesosphere/usi/examples/core-hello-world/target/scala-2.13/test-classes ...
[warn] /home/travis/build/mesosphere/usi/examples/core-hello-world/src/test/scala/com/mesosphere/usi/examples/CoreHelloWorldFrameworkTest.scala:110:17: parameter value existingFrameworkId in class Fixture is never used
[warn]   class Fixture(existingFrameworkId: Option[FrameworkID.Builder] = None) {
[warn]                 ^
[warn] one warning found
[info] Done compiling.
[info] Running scalafix on 2 Scala sources
[warn] there were 7 deprecation warnings (since 2.13.0)
[warn] there were two deprecation warnings (since 2.6.0)
[warn] there were 9 deprecation warnings in total; re-run with -deprecation for details
[warn] 5 warnings found
[info] Done compiling.
[info] Running scalafix on 14 Scala sources
[error] (commons / Test / compileIncremental) Compilation failed
[error] Total time: 24 s, completed Jun 25, 2020 3:25:54 PM

restoring stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
</pre>
</details>

**Comment by timcharper** [comment]
Exit status 1 when running "Running tests for all Scala versions"

<details>
<summary>Details</summary>
<pre>
[info] All tests passed.
[info] Run completed in 1 minute, 23 seconds.
[info] Total number of tests run: 1
[info] Suites: completed 1, aborted 0
[info] Tests: succeeded 1, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[info] Run completed in 1 minute, 18 seconds.
[info] Total number of tests run: 1
[info] Suites: completed 1, aborted 0
[info] Tests: succeeded 1, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[info] Run completed in 1 minute, 14 seconds.
[info] Total number of tests run: 8
[info] Suites: completed 2, aborted 0
[info] Tests: succeeded 8, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[error] (commons / Test / compileIncremental) Compilation failed
[error] Total time: 114 s (01:54), completed Jun 25, 2020 3:28:06 PM

restoring stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
</pre>
</details>

**Comment by timcharper** [comment]
Exit status 1 when running "sbt +test:compile"

<details>
<summary>Details</summary>
<pre>
[info] Done compiling.
[info] Compiling 1 Scala source to /home/travis/build/mesosphere/usi/examples/core-hello-world/target/scala-2.12/classes ...
[warn] there were two deprecation warnings (since 2.6.0); re-run with -deprecation for details
[warn] one warning found
[info] Done compiling.
[info] Compiling 1 Scala source to /home/travis/build/mesosphere/usi/mesos-master-detector/target/scala-2.12/test-classes ...
[info] Done compiling.
[warn] there were two deprecation warnings (since 2.6.0); re-run with -deprecation for details
[warn] three warnings found
[info] Done compiling.
[info] Compiling 2 Scala sources to /home/travis/build/mesosphere/usi/examples/core-hello-world/target/scala-2.12/test-classes ...
[warn] /home/travis/build/mesosphere/usi/examples/core-hello-world/src/test/scala/com/mesosphere/usi/examples/CoreHelloWorldFrameworkTest.scala:110:17: parameter value existingFrameworkId in class Fixture is never used
[warn]   class Fixture(existingFrameworkId: Option[FrameworkID.Builder] = None) {
[warn]                 ^
[warn] one warning found
[info] Done compiling.
[error] (commons / Test / compileIncremental) Compilation failed
[error] Total time: 86 s (01:26), completed Jun 25, 2020 4:02:05 PM

restoring stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
</pre>
</details>

**Comment by timcharper** [comment]
Exit status 1 when running "Unused imports / scalafix"

<details>
<summary>Details</summary>
<pre>
[warn] one warning found
[info] Done compiling.
[info] Running scalafix on 3 Scala sources
[info] Compiling 2 Scala sources to /home/travis/build/mesosphere/usi/examples/core-hello-world/target/scala-2.13/test-classes ...
[warn] /home/travis/build/mesosphere/usi/examples/core-hello-world/src/test/scala/com/mesosphere/usi/examples/CoreHelloWorldFrameworkTest.scala:110:17: parameter value existingFrameworkId in class Fixture is never used
[warn]   class Fixture(existingFrameworkId: Option[FrameworkID.Builder] = None) {
[warn]                 ^
[warn] there were 7 deprecation warnings (since 2.13.0)
[warn] there were two deprecation warnings (since 2.6.0)
[warn] there were 9 deprecation warnings in total; re-run with -deprecation for details
[warn] 5 warnings found
[info] Done compiling.
[info] Running scalafix on 14 Scala sources
[warn] one warning found
[info] Done compiling.
[info] Running scalafix on 2 Scala sources
[error] (commons / Test / compileIncremental) Compilation failed
[error] Total time: 26 s, completed Jun 25, 2020 4:04:25 PM

restoring stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
</pre>
</details>

**Comment by timcharper** [comment]
Exit status 1 when running "Running tests for all Scala versions"

<details>
<summary>Details</summary>
<pre>
[info] All tests passed.
[info] Run completed in 1 minute, 12 seconds.
[info] Total number of tests run: 8
[info] Suites: completed 2, aborted 0
[info] Tests: succeeded 8, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[info] Run completed in 1 minute, 19 seconds.
[info] Total number of tests run: 73
[info] Suites: completed 11, aborted 0
[info] Tests: succeeded 73, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[info] Run completed in 1 minute, 21 seconds.
[info] Total number of tests run: 1
[info] Suites: completed 1, aborted 0
[info] Tests: succeeded 1, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[error] (commons / Test / compileIncremental) Compilation failed
[error] Total time: 114 s (01:54), completed Jun 25, 2020 4:06:37 PM

restoring stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
</pre>
</details>

File: commons/src/main/scala/com/mesosphere/usi/async/ExecutionContexts.scala [hunk 2]
```diff
@@ -14,16 +23,21 @@ object ThreadPoolContext {
...
+    * An [[ExecutionContext]] that uses the same thread. Use it wisely.
+    */
+  lazy val callerThread: ExecutionContext = CallerThreadExecutionContext()
 
   /**
     * Returns an execution context backed by a fixed thread pool. All threads in the pool are prefixed with `namePrefix`
     * e.g. `slow-io-pool-thread-1`.
     *
-    * @param numThreads number of threads in the pool
+   * @param numThreads number of threads in the pool
```
Comment (pierrebeitz) [inline commons/src/main/scala/com/mesosphere/usi/async/ExecutionContexts.scala]: we could need a scalafmt ci-check :)

**Review by pierrebeitz — APPROVED** [review]
LGTM

File: commons/src/main/scala/com/mesosphere/usi/async/ExecutionContexts.scala (hunk 2 above)
Comment (jeschkies) [reply commons/src/main/scala/com/mesosphere/usi/async/ExecutionContexts.scala]: We do have it.

**Pull request merged**